rust_src/src/symbols.rs

```diff
@@ -487,6 +489,46 @@ pub fn local_variable_p(mut symbol: LispSymbolRef, buffer: LispBufferOrCurrent)
     }
 }
 
+/// Return a value indicating where VARIABLE's current binding comes from.
+/// If the current binding is buffer-local, the value is the current buffer.
+/// If the current binding is global (the default), the value is nil.
+#[lisp_fn]
+pub fn variable_binding_locus(mut symbol: LispSymbolRef) -> LispObject {
+    // Make sure the current binding is actually swapped in.
+    unsafe {
+        symbol.find_value();
+    }
+    symbol = symbol.get_indirect_variable();
+
+    match symbol.get_redirect() {
```

Comment:
Continuing my comment about `fallthrough` I think this has to be written as if/else statements with explicit returns OR the `symbol_localized` code is copied into the fallthrough else that is currently written. Either way, yuck.

Reply:
absorbing `symbol_localized` into an inner function looks somewhat cleaner.